Physics/BoundingShapes/ShapesExtractor.cs

```diff
@@ -0,0 +1,157 @@
+﻿using System.Numerics;
+using System.Runtime.CompilerServices;
+using BepuPhysics;
+using BepuPhysics.Collidables;
+using BepuUtilities.Collections;
+using BepuUtilities.Memory;
+using Physics.Collisions;
+
+namespace Physics.BoundingShapes;
+public class ShapesExtractor : IDisposable
+{
+    private readonly Simulation _simulation;
+
+    private readonly BufferPool _pool;
+
+    public QuickList<BoundingBox> Boxes;
```

Comment:
Why are we using this instead of some normal collection like List?

Reply:
Could'nt we just use list and remove all this unsafe stuff

Reply:
We can't get rid of unsafe completely because we need to make calls to low level bepu api which uses unsafe.
QuickLists were originally used in bepu & are faster than normal lists so I thought

Reply:
I think its less readable at its no like we need performance here. We only have 12 shapes at most i think